bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -4788,13 +4788,22 @@ void RewriteInstance::updateELFSymbolTable(
     if (!IsDynSym && shouldStrip(Symbol))
       continue;
 
+    Expected<StringRef> SymbolName = Symbol.getName(StringSection);
+    assert(SymbolName && "cannot get symbol name");
```

Comment:
is this assert needed? if Expected returns null, it will blow up, no?

Reply:
Not really. Let me drop it. Yes, Expected will blow up upon null access, and the assertion message is not more insightful than SymbolName variable itself.

Reply:
> Expected<T> values in success mode must still be checked prior to being destroyed

So we must keep the assertion.

Reply:
SG